File: MacOS/Kinogrida/Constants/KGGridConfig.swift
import AppKit

final class KGGridConfig {
    var nbrColumns: Int     = 10
    var nbrRows: Int        = 10
    var cellSize: CGFloat   = 1
    var gridMargin: CGFloat = 50
    var offsetX: CGFloat    = 0
    var offsetY: CGFloat    = 0
    var width: CGFloat      = 0
    var height: CGFloat     = 0
    var lineWidth: CGFloat  = 1
    var speed: CGFloat      = 10000
    var colors: [NSColor]   = []

    // ── Debug ─────────────────────────────────────────────────
    var showGrid: Bool        = false
    var showStats: Bool       = false
    var showLockedCells: Bool = false
    var showPath: Bool        = false
    var showPosition: Bool    = false

    static func defaultConfig() -> KGGridConfig {
        let c    = KGGridConfig()
        c.colors = KGColors.randomPalette()
        return c
    }

    // Recompute cellSize, offsets, lineWidth from canvas bounds
    func updateForCanvasSize(_ size: CGSize) {
        let availW  = size.width  - 2 * gridMargin
        let availH  = size.height - 2 * gridMargin
        cellSize    = floor(min(availW / CGFloat(nbrColumns), availH / CGFloat(nbrRows)))
        width       = CGFloat(nbrColumns) * cellSize
        height      = CGFloat(nbrRows)    * cellSize
        offsetX     = (size.width  - width)  / 2
        offsetY     = (size.height - height) / 2
        lineWidth   = max(1, floor(cellSize * 0.1))
    }
}

File: Windows/Engine/KGGridConfig.cs
using System;
using System.Drawing;

namespace Kinogrida
{
    class KGGridConfig
    {
        public int NbrColumns = 10;
        public int NbrRows    = 10;
        public float CellSize  = 1f;
        public float GridMargin = 50f;
        public float OffsetX   = 0f;
        public float OffsetY   = 0f;
        public float Width     = 0f;
        public float Height    = 0f;
        public float LineWidth = 1f;
        public float Speed     = 10000f;
        public Color[] Colors  = Array.Empty<Color>();

        public static KGGridConfig Default()
        {
            var c = new KGGridConfig();
            c.Colors = KGColors.RandomPalette();
            return c;
        }

        public void UpdateForCanvasSize(float canvasW, float canvasH)
        {
            float availW = canvasW - 2 * GridMargin;
            float availH = canvasH - 2 * GridMargin;
            CellSize = MathF.Floor(MathF.Min(availW / NbrColumns, availH / NbrRows));
            Width    = NbrColumns * CellSize;
            Height   = NbrRows    * CellSize;
            OffsetX  = (canvasW - Width)  / 2f;
            OffsetY  = (canvasH - Height) / 2f;
            LineWidth = MathF.Max(1f, MathF.Floor(CellSize * 0.1f));
        }
    }
}
